classes/requests/helpers/class-paysoncheckout-for-woocommerce-helper-gui.php

```diff
@@ -33,6 +33,11 @@ public function get_gui() {
 	public function get_payson_language() {
 		$iso_code      = explode( '_', get_locale() );
 		$shop_language = $iso_code[0];
+
+		if ( function_exists( 'weglot_get_current_language' ) ) {
+			$shop_language = weglot_get_current_language();
+		}
```

Comment:
Has this been discussed with weglot at all? This is not something that should be handled by us as a payment method, but rather something Weglot should set properly so that get_locale() returns the locale they have set for the site.